File: 33.js
function groupAnagrams(words) {
  const map = new Map();

  for (let word of words) {
    // Sort the word
    const key = word.split('').sort().join('');
    if (!map.has(key)) {
      map.set(key, []);
    }
    map.get(key).push(word);
  }

  return Array.from(map.values());
}

// --- Example ---
let words = ["eat", "tea", "tan", "ate", "nat", "bat"];
console.log(groupAnagrams(words));

File: 33.py
from collections import defaultdict

def group_anagrams(words):
    groups = defaultdict(list)

    for word in words:
        # Sort the word to form a key
        key = "".join(sorted(word))
        groups[key].append(word)

    return list(groups.values())


# --- Example ---
words = ["eat", "tea", "tan", "ate", "nat", "bat"]
print(group_anagrams(words))
